### PR TITLE
 feat: create UI for seat selection

### PR DESCRIPTION
**Pull request by TonNQ**
Show seat map after selecting showtime and show seats selected by user

86ept9djf

### CODE REVIEW

File: src/components/Seat/Seat.jsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import { cn } from '@/lib/utils'
+import SingleSeat from '@/assets/images/seat-single.svg'
```
Comment (ces-anhlam) [inline src/components/Seat/Seat.jsx:2]: import as component

File: src/redux/slices/bookTicket.js [hunk 2]
```diff
@@ -19,6 +21,16 @@ const bookTicketSlice = createSlice({
     setSelectedCinemaShow: (state, action) => {
       state.selectedCinemaShow = action.payload
     },
+    toggleSelectedSeat: (state, action) => {
+      const { name } = action.payload
+      const seatIndex = state.seatList.findIndex((seat) => seat.name === name)
+
+      if (seatIndex !== -1) {
```
Comment (ces-anhlam) [inline src/redux/slices/bookTicket.js:28]: don't use magic number

File: src/redux/slices/bookTicket.js [hunk 3]
```diff
@@ -30,7 +42,37 @@ const bookTicketSlice = createSlice({
...
+export const countSelectedSeats = createSelector(
+  (state) => state.bookTicket.seatList,
+  (seatList) => {
+    return seatList ? seatList.length : 0
+  }
+)
+
+export const sortedSelectedSeatList = createSelector(
+  (state) => state.bookTicket.seatList,
+  (seatList) => [...seatList].sort((a, b) => (a.name > b.name ? 1 : -1))
```
Comment (ces-anhlam) [inline src/redux/slices/bookTicket.js:56]: same above

Comment (TonNQ) [reply src/redux/slices/bookTicket.js:56]: fixed

File: src/redux/slices/bookTicket.js (hunk 2 above)
Comment (TonNQ) [reply src/redux/slices/bookTicket.js:28]: fixed

File: src/components/Seat/Seat.jsx (hunk 1 above)
Comment (TonNQ) [reply src/components/Seat/Seat.jsx:2]: fixed

File: package.json [hunk 4]
```diff
@@ -12,8 +12,8 @@
     "format": "prettier --write './**/*.{js,jsx,ts,tsx,css,md,json}' --config ./.prettierrc"
   },
   "dependencies": {
-    "@reduxjs/toolkit": "^2.2.6",
     "@tailwindcss/aspect-ratio": "^0.4.2",
+    "@reduxjs/toolkit": "^2.2.6",
```
Comment (ces-anhlam) [inline package.json:16]: why did you change position of package?

File: src/utils/price.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
+export const formatPrice = (price, decimalSeparator = '.', thousandSeparator = ',') => {
```
Comment (ces-anhlam) [inline src/utils/price.js:1]: I don't see you use  this function

File: src/components/MovieInfoTotal/MovieInfoTotal.jsx [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+  const handleClose = () => {
+    dispatch(clearShowtime())
+  }
+
+  const handleSubmitSeats = async () => {
+    const currentTime = moment().format()
+    try {
+      await setSeatsOnHold(
+        selectedShowtime.showId,
+        seatList.map((seat) => ({
```
Comment (ces-anhlam) [inline src/components/MovieInfoTotal/MovieInfoTotal.jsx:34]: ```seatList.map(({colId, name}) => ...```

File: src/components/Seat/Seat.jsx [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+    dispatch(toggleSelectedSeat(seat))
+  }
+
+  return (
+    <div
+      className={cn('relative flex min-h-[30px] min-w-[40px] items-center justify-center', {
+        'hover:cursor-pointer': seat.isSeat && !seat.isSold,
+        'hover:cursor-default': !seat.isSeat || seat.isSold || seat.onHold
+      })}
+      onClick={seat.isSold || seat.onHold ? null : toggleSelected}
```
Comment (ces-anhlam) [inline src/components/Seat/Seat.jsx:21]: destructuring

**Review by ces-anhlam — APPROVED** [review]

**Review by ces-khanhhoang — APPROVED** [review]

**Comment by TonNQ** [comment]
I just updated API following BE and didn't affect others
